### PR TITLE
Put create/update/delete behind admin role

### PR DESCRIPTION
**Pull request by samleeflang**
Small change in how we get the authorities (same as roles).
We can now also use @PreAuthorize("hasRole('orchestration-admin')") if we want on specific controller methods.
Only included roles from the `resource_access` are added.
We also got realm_access but these are mainly for Keycloak authorisation don't think we need them.

@TomDijkema added you so you see what we do in the backend. Hopefully it gives some inspiration for the Frontend :wink:

### CODE REVIEW

File: src/main/java/eu/dissco/orchestration/backend/security/JwtAuthConverter.java [hunk 1]
```diff
@@ -2,33 +2,39 @@
...
+      ((Map<String, Object>) jwt.getClaims().get("resource_access")).forEach((k, v) -> {
+        Map<String, Object> resourceAccess = (Map<String, Object>) v;
+        resourceAccess.forEach((k1, v1) -> {
+          if (k1.equals("roles")) {
+            ((Collection<String>) v1).forEach(
+                role -> authorities.add((GrantedAuthority) () -> "ROLE_" + role));
+          }
+        });
+      });
+    }
```
Comment (southeo) [inline src/main/java/eu/dissco/orchestration/backend/security/JwtAuthConverter.java:36]: Not sure how we're assigning roles, but would it be noteworthy to log if a user has no roles? Basically, will a regular user have a "regular user" role, or will they have no roles?

File: src/main/java/eu/dissco/orchestration/backend/security/JwtAuthConverter.java [hunk 2]
```diff
@@ -2,33 +2,39 @@
...
-        .flatMap(Collection::stream);
+  private Set<GrantedAuthority> extractRoles(Jwt jwt) {
+    Set<GrantedAuthority> authorities = new HashSet<>();
+    if (jwt.getClaims().containsKey("resource_access")) {
+      ((Map<String, Object>) jwt.getClaims().get("resource_access")).forEach((k, v) -> {
+        Map<String, Object> resourceAccess = (Map<String, Object>) v;
+        resourceAccess.forEach((k1, v1) -> {
+          if (k1.equals("roles")) {
+            ((Collection<String>) v1).forEach(
+                role -> authorities.add((GrantedAuthority) () -> "ROLE_" + role));
```
Comment (southeo) [inline src/main/java/eu/dissco/orchestration/backend/security/JwtAuthConverter.java:32]: What is this lambda doing? is it constructing a new GrantedAuthority? I haven't seen syntax like this before.

File: src/test/java/eu/dissco/orchestration/backend/security/JwtAuthConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            "resource_access", Map.of("orchestration-service",
+                Map.of("roles", List.of("orchestration-admin")))));
+
+    // When
+    var token = converter.convert(jwt);
+
+    // Then
+    assertThat(token.isAuthenticated()).isTrue();
+    assertThat(token.getName()).isEqualTo("adf294ba-bb03-4962-8042-a37f1648458e");
+    assertThat(token.getAuthorities()).hasSize(1);
```
Comment (southeo) [inline src/test/java/eu/dissco/orchestration/backend/security/JwtAuthConverterTest.java:55]: Nice tests. Wasn't sure how to test the jwtConverter.

**Review by southeo — APPROVED** [review]
Few questions but they're just for clarity.
Looks good and straightforward, really happy to be getting roles into our system.

File: src/main/java/eu/dissco/orchestration/backend/security/JwtAuthConverter.java (hunk 1 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/orchestration/backend/security/JwtAuthConverter.java:36]: They might have no roles and I think that is ok. So they can be authenticated and have no additional roles.

File: src/main/java/eu/dissco/orchestration/backend/security/JwtAuthConverter.java (hunk 2 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/orchestration/backend/security/JwtAuthConverter.java:32]: Oh yeah, this was actually done by Copilot and it seems that a GrantedAuthority is an interface which you initiate through a function. Bit new for me as well, probably comes from the functional programming movement. Anyway was difficult to test as well as you cannot check the roleName as this is a function. Two functions are not equal as they are objects... Anyway the test now assumes the there is one GrantedAuthority and doesn't check the name... Crazy times

**Pull request merged**